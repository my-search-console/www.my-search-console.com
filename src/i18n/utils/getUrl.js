const stats = require("../urls/stats.json")
const leaderboard = require("../urls/leaderboard.json")
const lessons = require("../urls/lessons.json")
const levels = require("../urls/levels.json")
const share = require("../urls/share.json")
const signin = require("../urls/signin.json")
const terms = require("../urls/terms.json")
const privacy = require("../urls/privacy.json")

const getSigninUrl = (lang) => {
  if (lang === "en") return `/${signin[lang].signin}/`

  return `/${lang}/${signin[lang].signin}/`
}

exports.getSigninUrl = getSigninUrl

exports.getSigninLinkValidationUrl = (lang) => {
  return `${getSigninUrl(lang)}${signin[lang].validation}/`
}

exports.getLeaderboardUrl = (lang) => {
  if (lang === "en") return `/${leaderboard[lang]}/`

  return `/${lang}/${leaderboard[lang]}/`
}

exports.navigateToStats = (lang) => {
  if (lang === "en") return `/${stats[lang]}/`

  return `/${lang}/${stats[lang]}/`
}

exports.getShareUrl = (lang) => {
  if (lang === "en") return `/${share[lang]}/`

  return `/${lang}/${share[lang]}/`
}

exports.getTermsOfUseUrl = (lang) => {
  if (lang === "en") return `/${terms[lang]}/`

  return `/${lang}/${terms[lang]}/`
}

exports.getPrivacyUrl = (lang) => {
  if (lang === "en") return `/${privacy[lang]}/`

  return `/${lang}/${privacy[lang]}/`
}

exports.navigateToLessons = (lang) => {
  return lang === "en"
    ? `/${lessons[lang].index}/`
    : `/${lang}/${lessons[lang].index}/`
}

exports.getLessonUrl = ({ lang, level, url }) => {
  return lang === "en"
    ? `/${lessons[lang].index}/${levels[lang][level]}/${url}/`
    : `/${lang}/${lessons[lang].index}/${levels[lang][level]}/${url}/`
}

exports.getCourseUrl = ({ lang, level }) => {
  return lang === "en"
    ? `/${lessons[lang].index}/${levels[lang][level]}/`
    : `/${lang}/${lessons[lang].index}/${levels[lang][level]}/`
}

exports.getCoursesUrl = ({ lang }) => {
  return lang === "en"
    ? `/${lessons[lang].index}/`
    : `/${lang}/${lessons[lang].index}/`
}

exports.navigateToHome = (lang) => {
  return lang === "en" ? "/" : `/${lang}/`
}
