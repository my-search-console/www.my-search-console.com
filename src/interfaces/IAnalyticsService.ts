import { AnalyticsEntity } from "../entities/AnalyticsEntity"

export interface IAnalyticsService {
  logout(): Promise<void>
  authenticate(params: { id: string; created_at: Date | string }): Promise<void>
  send(analytics: AnalyticsEntity): any
}
