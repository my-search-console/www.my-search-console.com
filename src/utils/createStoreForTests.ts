import { init as createStore } from "../redux/store"
import { actions } from "../redux/actions"
import { TestModule } from "../modules/TestModule"

export const createStoreForTests = (initialState = {}) => {
  const di = new TestModule().build()

  const { store } = createStore(initialState, di)

  return {
    store,
    actions,
    di,
  }
}
