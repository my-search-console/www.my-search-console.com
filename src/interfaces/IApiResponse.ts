import { ErrorEntity } from "@my-search-console/interfaces"
import { InternalErrorEntity } from "../entities/InternalErrorEntity"

export type IApiResponse<T> =
  | ({ statusCode: 200 } & T)
  | { statusCode: 400; message: ErrorEntity }

export type IRepositoryResponse<T> =
  | ({ error: false } & { body: T })
  | { error: true; code: ErrorEntity | InternalErrorEntity; data?: any }
