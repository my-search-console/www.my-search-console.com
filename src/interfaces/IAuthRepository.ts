import {
  ErrorEntity,
  UserEntity,
  UserToGoogleSearchConsoleWithEmailsEntity,
} from "@foudroyer/interfaces"
import { IRepositoryResponse } from "./IApiResponse"

export type GetUserInfoResponse = IRepositoryResponse<UserEntity>
export type GetSourcesResponse = IRepositoryResponse<{
  bing: boolean
  google: boolean
  yandex: boolean
}>

export type AuthenticateWithGoogleResponse = IRepositoryResponse<{
  token: string
}>
export type AuthenticateWithYandexResponse = IRepositoryResponse<any>
export type GoogleSearchAccountGetParentsResponse = IRepositoryResponse<
  UserToGoogleSearchConsoleWithEmailsEntity[]
>
export type GoogleSearchAccountDeleteResponse = IRepositoryResponse<any>

export interface IAuthRepository {
  authenticateWithGoogle(params?: {
    language: string
  }): Promise<AuthenticateWithGoogleResponse>

  postAuthenticationCode(params: {
    code: string
    callbackUrl: string
    type: "google" | "yandex" | "bing"
    language?: string
  }): Promise<
    | {
        error: true
        code: ErrorEntity
      }
    | {
        error: false
        body: string
      }
  >

  /**
   *
   * Google Search Console
   *
   */
  addGoogleSearchAccount(params?: {
    language: string
  }): Promise<AuthenticateWithGoogleResponse>

  GoogleSearchAccountGetParents(): Promise<GoogleSearchAccountGetParentsResponse>
  GoogleSearchAccountDelete(params: {
    id: UserToGoogleSearchConsoleWithEmailsEntity["id"]
  }): Promise<GoogleSearchAccountGetParentsResponse>

  /**
   *
   *
   *
   * ----------
   *
   *
   *
   *
   */

  authenticateWithYandex(): Promise<AuthenticateWithYandexResponse>
  authenticateWithBing(): Promise<AuthenticateWithYandexResponse>
  getUserInfo(): Promise<GetUserInfoResponse>
  getSources(): Promise<GetSourcesResponse>
}
