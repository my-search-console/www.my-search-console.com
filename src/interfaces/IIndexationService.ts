import { IndexationGoogleCloudApiKeyEntity } from "@my-search-console/interfaces"
import { IndexationSearchEngines } from "../entities/SearchEngineEntity"
import { IApiResponse, IRepositoryResponse } from "./IApiResponse"

export type IndexResponseType = IRepositoryResponse<any>
export type CheckIndexNowKeyResponse = IRepositoryResponse<{
  success: true
}>

export type GetGoogleApiKeysResponse = IRepositoryResponse<{
  google_cloud_api_keys: IndexationGoogleCloudApiKeyEntity[]
}>

export type AddGoogleApiKeyResponse = IRepositoryResponse<any>
export type DeleteGoogleApiKeyResponse = IRepositoryResponse<any>

export interface IIndexationService {
  checkIndexNowKey(params: {
    websiteId: string
  }): Promise<CheckIndexNowKeyResponse>
  index(params: {
    url: string
    websiteId: string
    types: IndexationSearchEngines[]
    mode: "queue" | "manual"
  }): Promise<IndexResponseType>

  deleteGoogleApiKey(params: {
    keyId: string
    websiteId: string
  }): Promise<DeleteGoogleApiKeyResponse>

  downloadGoogleApiKey(params: {
    google_cloud_api_key: string
  }): Promise<IApiResponse<any>>

  getGoogleApiKeys(params: {
    websiteId: string
  }): Promise<GetGoogleApiKeysResponse>

  addGoogleApiKey(params: {
    websiteId: string
    key: string
  }): Promise<AddGoogleApiKeyResponse>
}
