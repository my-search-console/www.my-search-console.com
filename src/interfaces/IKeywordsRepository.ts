import { RankingStatEntity } from "./../entities/RankingWebsiteEntity"
import { IRepositoryResponse } from "./IApiResponse"

export interface IKeywordsRepository {
  fetch(params: {
    website: string
    filter?: {
      country?: string | null
      device?: "unknown" | "desktop" | "mobile" | "tablet" | null
      source?: "google" | "bing" | "yandex" | null
    }
  }): Promise<IRepositoryResponse<RankingStatEntity[]>>
  create(params: {
    websiteId: string
    keywords: string[]
  }): Promise<IRepositoryResponse<null>>
  delete(params: {
    websiteId: string
    keywords: string[]
  }): Promise<IRepositoryResponse<null>>
}
