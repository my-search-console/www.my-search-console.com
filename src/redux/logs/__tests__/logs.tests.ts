import { LogsEntity } from "@foudroyer/interfaces"
import { WebsiteActivated } from "../../../modules/seeds/WebsitesSeeds"
import { createStoreForTests } from "../../../utils/createStoreForTests"

describe("logs tests suite", () => {
  it("le reducer initial doit être conforme au snapshot", async () => {
    const { store, actions, di } = await createStoreForTests()
    expect(store.getState().logs).toMatchSnapshot()
  })

  it("je dois pouvoir fetch les logs", async () => {
    const { store, actions, di } = await createStoreForTests({
      seeds: true,
    })

    di.LogsRepository.createOrUpdate({
      id: "id",
      fk_website_id: WebsiteActivated.id,
      title: "",
      description: "",
      log_date: new Date(),
      created_at: new Date(),
      updated_at: new Date(),
      clicks: 0,
      impressions: 0,
      type: "global",
      query: null,
      page: null,
    })

    expect(store.getState().logs.logs).toHaveLength(0)

    await store.dispatch<any>(actions.logs.$fetch())

    expect(store.getState().logs.logs).toHaveLength(1)
  })

  it("si je crée un log, le titre est obligatoire", async () => {
    const { store, actions, di } = await createStoreForTests({
      seeds: true,
    })

    const log: LogsEntity = {
      id: "",
      fk_website_id: "",
      title: "",
      description: "",
      log_date: new Date(),
      created_at: new Date(),
      updated_at: new Date(),
      clicks: 0,
      impressions: 0,
      type: "global",
      query: null,
      page: null,
    }

    await store.dispatch<any>(actions.logs.$create(log))

    expect(store.getState().logs.logs).toHaveLength(0)
  })

  it("je dois pouvoir créer un log", async () => {
    const { store, actions, di } = await createStoreForTests({
      seeds: true,
    })

    expect(store.getState().logs.logs).toHaveLength(0)

    const log: LogsEntity = {
      id: "aziadoijz",
      fk_website_id: WebsiteActivated.id,
      title: "hello",
      description: "",
      log_date: new Date(),
      created_at: new Date(),
      updated_at: new Date(),
      clicks: 0,
      impressions: 0,
      type: "global",
      query: null,
      page: null,
    }

    await store.dispatch<any>(actions.logs.$fetch())

    expect(store.getState().logs.logs).toHaveLength(0)

    await store.dispatch<any>(actions.logs.$create(log))

    expect(store.getState().logs.logs).toHaveLength(1)
    expect(store.getState().logs.logs[0].created_at).toEqual(log.created_at)
  })

  it("je dois pouvoir supprimer un log", async () => {
    const { store, actions, di } = await createStoreForTests({
      seeds: true,
    })

    expect(store.getState().logs.logs).toHaveLength(0)

    const log: LogsEntity = {
      id: "id",

      fk_website_id: WebsiteActivated.id,
      title: "hello",
      description: "",
      log_date: new Date(),
      created_at: new Date(),
      updated_at: new Date(),
      clicks: 0,
      impressions: 0,
      type: "global",
      query: null,
      page: null,
    }

    await store.dispatch<any>(actions.logs.$create(log))
    await store.dispatch<any>(actions.logs.$delete(log.id))

    expect(store.getState().logs.logs).toHaveLength(0)
  })

  it("je dois pouvoir modifier un log", async () => {
    const { store, actions, di } = await createStoreForTests({
      seeds: true,
    })

    expect(store.getState().logs.logs).toHaveLength(0)

    const log: LogsEntity = {
      id: "id",
      fk_website_id: WebsiteActivated.id,
      title: "hello",
      description: "",
      log_date: new Date(),
      created_at: new Date(),
      updated_at: new Date(),
      clicks: 0,
      impressions: 0,
      type: "global",
      query: null,
      page: null,
    }

    await store.dispatch<any>(actions.logs.$create(log))
    await store.dispatch<any>(
      actions.logs.$update({ ...log, title: "changed" })
    )

    expect(store.getState().notifications.notifications.length).toEqual(0)
    expect(store.getState().logs.logs.length).toEqual(1)
    expect(store.getState().logs.logs[0].title).toEqual("changed")
  })
})
