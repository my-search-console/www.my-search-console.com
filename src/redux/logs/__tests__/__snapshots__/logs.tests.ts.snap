// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs tests suite le reducer initial doit Ãªtre conforme au snapshot 1`] = `
{
  "createOrUpdateForm": {
    "form": null,
    "isFetching": false,
  },
  "isFetching": false,
  "logs": [],
  "tags": [],
}
`;
